client/connector_create.go

```diff
@@ -42,6 +42,17 @@ func generateConnectorName(namespace string, cli kubernetes.Interface) string {
 	return "conn" + strconv.Itoa(max)
 }
 
+func (cli *VanClient) isOwnToken(ctx context.Context, secret *corev1.Secret) (bool, error) {
+	siteConfig, err := cli.SiteConfigInspect(ctx, nil)
+	if err != nil {
+		return false, err
+	}
+	if author, ok := secret.ObjectMeta.Annotations[types.TokenGeneratedBy]; ok {
+		return author == siteConfig.Reference.UID, nil
+	}
+	return false, nil
+}
+
 func (cli *VanClient) ConnectorCreateFromFile(ctx context.Context, secretFile string, options types.ConnectorCreateOptions) (*corev1.Secret, error) {
 	secret, err := cli.ConnectorCreateSecretFromFile(ctx, secretFile, options)
```

Comment:
Rather than create and then immediately delete a resource, would lean towards inspecting the token first and ownToken check before create.

In pkg/certs/certs.go  there is a 'GetSecretContent' that will return the data and annotations to check against as a map[string] e.g. content["skupper.io/generated-by"]

Reply:
Ah, thanks. I should have seen that.